src/main/java/org/polypheny/jdbc/meta/PolyphenyColumnMeta.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.jdbc.meta;
+
+import java.sql.JDBCType;
+import java.sql.ResultSetMetaData;
+import lombok.Getter;
+import org.apache.commons.lang3.NotImplementedException;
+import org.polypheny.prism.ColumnMeta;
+import org.polypheny.prism.ProtoPolyType;
+import org.polypheny.jdbc.types.ProtoToJdbcTypeMap;
+
+public class PolyphenyColumnMeta {
+
+    @Getter
+    private final int ordinal;
+    @Getter
+    private final boolean autoIncrement;
+    @Getter
+    private final boolean caseSensitive;
+    @Getter
+    private final boolean searchable;
+    @Getter
+    private final boolean currency;
+    @Getter
+    private final int nullable;
+    @Getter
+    private final boolean signed;
+    @Getter
+    private final int displaySize;
+    @Getter
+    private final String columnLabel;
+    @Getter
+    private final String columnName;
+    @Getter
+    private final String namespace;
+    @Getter
+    private final int precision;
+    @Getter
+    private final int scale;
+    @Getter
+    private final String tableName;
+    @Getter
+    private final String catalogName;
+    @Getter
+    private final boolean readOnly;
+    @Getter
+    private final boolean writable;
+    @Getter
+    private final boolean definitelyWritable;
+    @Getter
+    private final String columnClassName;
+    @Getter
+    private final int sqlType;
+    @Getter
+    private final String polyphenyFieldTypeName;
+
+
+    //column = field
+    public PolyphenyColumnMeta( ColumnMeta protoColumnMeta ) {
+        this.ordinal = protoColumnMeta.getColumnIndex();
+        this.autoIncrement = false;
+        this.caseSensitive = true;
+        this.searchable = false;
+        this.currency = false;
+        this.nullable = protoColumnMeta.getIsNullable() ? ResultSetMetaData.columnNullable : ResultSetMetaData.columnNoNulls;
+        this.signed = false;
+        this.displaySize = protoColumnMeta.getLength();
+        this.columnLabel = protoColumnMeta.getColumnLabel();
+        this.columnName = protoColumnMeta.getColumnName();
+        this.namespace = protoColumnMeta.getNamespace();
+        this.precision = protoColumnMeta.getPrecision();
+        this.scale = 1;
+        // table = entity
+        this.tableName = protoColumnMeta.getEntityName();
+        this.catalogName = "";
+        this.readOnly = false;
+        this.writable = false;
+        this.definitelyWritable = false;
+        this.columnClassName = "";
+        if ( protoColumnMeta.getTypeMeta().getProtoValueType() == ProtoPolyType.USER_DEFINED_TYPE ) {
+            //TODO handle structured meta
```

Comment:
ToDo?

Reply:
This is required once user defined types are fully implemented on both the driver and the server. Depending on their implementation this might even become obsolete. This is something to address in the future.